Sources/Filters+Strings+Lettercase.swift

```diff
@@ -0,0 +1,157 @@
+            CharacterSet.lowercaseLetters.contains(scalar) {
+            idx = scalars.index(before: idx)
+        }
+        let transformed = String(scalars[start..<idx]).lowercased() + String(scalars[idx..<scalars.endIndex])
+        return transformed
+    }
+
+    /// Lowers the first letter of the string
+    /// e.g. "People picker" gives "people picker", "Sports Stats" gives "sports Stats"
+    ///
+    /// - Parameters:
+    ///   - value: the value to uppercase first letter of
+    ///   - arguments: the arguments to the function; expecting zero
+    /// - Returns: the string with first letter being uppercased
+    /// - Throws: FilterError.invalidInputType if the value parameter isn't a string
+    static func upperFirstLetter(_ value: Any?) throws -> Any? {
+        guard let string = value as? String else { throw Filters.Error.invalidInputType }
+        return titlecase(string)
+    }
+
+    /// Converts snake_case to camelCase. Takes an optional Bool argument for removing any resulting
+    /// leading '_' characters, which defaults to false
+    ///
+    /// - Parameters:
+    ///   - value: the value to be processed
+    ///   - arguments: the arguments to the function; expecting zero or one boolean argument
+    /// - Returns: the camel case string
+    /// - Throws: FilterError.invalidInputType if the value parameter isn't a string
+    static func snakeToCamelCase(_ value: Any?, arguments: [Any?]) throws -> Any? {
+        let stripLeading = try Filters.parseBool(from: arguments, required: false) ?? false
+        guard let string = value as? String else { throw Filters.Error.invalidInputType }
+
+        return try snakeToCamelCase(string, stripLeading: stripLeading)
+    }
+
+    /// Converts camelCase to snake_case. Takes an optional Bool argument for making the string lower case,
+    /// which defaults to true
+    ///
+    /// - Parameters:
+    ///   - value: the value to be processed
+    ///   - arguments: the arguments to the function; expecting zero or one boolean argument
+    /// - Returns: the snake case string
+    /// - Throws: FilterError.invalidInputType if the value parameter isn't a string
+    static func camelToSnakeCase(_ value: Any?, arguments: [Any?]) throws -> Any? {
+        let toLower = try Filters.parseBool(from: arguments, required: false) ?? true
+        guard let string = value as? String else { throw Filters.Error.invalidInputType }
+
+        let snakeCase = try snakecase(string)
+        if toLower {
+            return snakeCase.lowercased()
+        }
+        return snakeCase
+    }
+
+    /// Converts snake_case to camelCase, stripping prefix underscores if needed
+    ///
+    /// - Parameters:
+    ///   - string: the value to be processed
+    ///   - stripLeading: if false, will preserve leading underscores
+    /// - Returns: the camel case string
+    static func snakeToCamelCase(_ string: String, stripLeading: Bool) throws -> String {
```

Comment:
This should probably be moved to private, except that swiftIdentifier depends on it.

Should we have a clearer separation between the Stencil filter functions (that accept and return Any), and the actual filter methods (that accept and return String + typed arguments)?